CoursesParser/BaseParser.cs

```diff
@@ -0,0 +1,188 @@
+using HtmlAgilityPack;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Text;
+
+
+namespace CoursesParser
+{
+    public class BaseParser
+    {
+        private HtmlDocument _document;
+        private HtmlNode _selectFaculties;
+        private HtmlNode _selectTown;
+        private HtmlNode _selectStudyType;
+        private HtmlNode _selectStudyYear;
+        private HtmlNode _selectFieldOfStudy;
+        private HtmlNode _selectFieldOfStudyDetailed;
+        private HtmlNode _selectPlansTable;
+        private string _facId;
+        private string _townId;
+        private string _studyTypeId;
+
+        
+        private string _studyYearId;
+        private string _fieldOfStudyId;
+        private string _fieldOfStudyDetailedId;
+        private HashSet<string> _allCourses;
+        private Encoding _encoding;
+
+        public BaseParser()
+        {
+            _encoding = CodePagesEncodingProvider.Instance.GetEncoding(1250);
+            var url = "https://vzdelavanie.uniza.sk/vzdelavanie/plany.php";
+            var web = new HtmlWeb();
+            web.AutoDetectEncoding = false;
+
+            web.OverrideEncoding = _encoding;
+
+            _allCourses = new HashSet<string>(5000);
+            _document = web.Load(url);
+            _document.OptionDefaultStreamEncoding = _encoding;
+
+            _selectFaculties = _document.GetElementbyId("f");
+            _selectTown = _document.GetElementbyId("t");
+            _selectStudyType = _document.GetElementbyId("m");
+            _selectStudyYear = _document.GetElementbyId("r");
+            _selectFieldOfStudy = _document.GetElementbyId("o");
+            _selectFieldOfStudyDetailed = _document.GetElementbyId("z");
+            _selectPlansTable = _document.GetElementbyId("plany").ChildNodes["table"];
+
+        }
+
+
+        public HashSet<string> ParseFaculties()
+        {
+            
+            foreach (var facultyOption in _selectFaculties.ChildNodes)
+            {
+                _facId = facultyOption.Attributes["value"].Value;
+                DownloadAndSaveCourses(1);
```

Comment:
What does it mean 1,2, 3... May be enum would be more clear...

Reply:
Is is simulating magic javascript function on the website vzdelavanie - I replace it with Enum